ex04.ex

```diff
@@ -75,7 +75,12 @@ defmodule Ex04 do
   return value will be the thing you have to manipulate.
   """
 
-  def even_odd . . . "your code"
+
```

Comment:
This is a nice solution.

I'm not going to deduct anything, because I haven't mentioned this, but in generate it is better to use [a|tail] to build a list, rather than ++, as it avoids a full list copy. You'll need to reverse either the input or the results, but it is still considered to be more efficient.

~~~ elixir
  def even_odd(list) do
    list
    |> reverse()
    |> reduce({ [], [] }, &partition_even_odd/2)
  end

  defp partition_even_odd(x, { es, os }) when Integer.is_even(x), do: {[x | es], os}
  defp partition_even_odd(x, { es, os }) when Integer.is_odd(x),  do: {es, [x | os]}
~~~